### ISSUE TITLE
g-w: cache misses should not be fatal

### ISSUE THREAD

**Issue by aerickson** [taskcluster/taskcluster#7163]
**Describe the bug**
Some of our hardware host's disks were getting full (https://bugzilla.mozilla.org/show_bug.cgi?id=1774954), so we cleaned up /opt/worker/downloads. This seems to have causes some exceptions (see https://bugzilla.mozilla.org/show_bug.cgi?id=1908736). Ideally, cache misses wouldn't be fatal (can't they just be fetched?).

**Expected behavior**
g-w would fetch the file missing from cache from the network.

**Taskcluster version**

```
[taskcluster 2024-07-18T16:06:40.357Z]     "generic-worker": {
[taskcluster 2024-07-18T16:06:40.357Z]       "engine": "simple",
[taskcluster 2024-07-18T16:06:40.357Z]       "go-arch": "amd64",
[taskcluster 2024-07-18T16:06:40.357Z]       "go-os": "darwin",
[taskcluster 2024-07-18T16:06:40.357Z]       "go-version": "go1.22.0",
[taskcluster 2024-07-18T16:06:40.357Z]       "release": "https://github.com/taskcluster/taskcluster/releases/tag/v60.3.4",
[taskcluster 2024-07-18T16:06:40.357Z]       "revision": "943a6f2b0d14fa0270280bc6f23acc2945d0fe45",
[taskcluster 2024-07-18T16:06:40.357Z]       "source": "https://github.com/taskcluster/taskcluster/commits/943a6f2b0d14fa0270280bc6f23acc2945d0fe45",
[taskcluster 2024-07-18T16:06:40.357Z]       "version": "60.3.4"
```

**Comment by matt-boris** [comment]
I'm surprised you hit this and not https://github.com/taskcluster/taskcluster/blob/49dd448aab512a1218907714e9a0a445ee1265b0/workers/generic-worker/mounts_feature.go#L174 which results in the worker refetching, as you suggested in your issue. I'd have to dig around more to see why yours crashed the worker. But if you remove your cache/download dirs, then I'd recommend just blowing away their respective, `file-caches.json`/`directory-caches.json` as well at that point.

**Comment by aerickson** [comment]
Yeah, this was awhile back... might not be possible to hit any longer / in a modern build. 

We've been bad about updating g-w on these also.
